### ISSUE TITLE
Migrate all the pipelines out of Prow

### ISSUE THREAD

**Issue by janmedrek** [kyma-project/lifecycle-manager#1811]
### Description

All the pipelines (including the build ones) should be migrated to the GH Actions feature.

The list of pipelines that are still in Prow is available [here](https://github.com/kyma-project/test-infra/issues/11711).

The `release-template-operator-assets` is a bit problematic since we cannot upload the files ourselves to the productive storage. We need to align with Neighbors on how to proceed and log a follow-up ticket.

### Reasons

Prow will be sunset and we need to move to our own solution.

### Acceptance Criteria

- [x] All the necessary pipelines migrated to GH Actions
- [x] Jellyfish team owns no pipelines in Prow (besides `release-template-operator-assets`)
  - `release-template-operator-assets` get removed https://github.com/kyma-project/test-infra/pull/11771
- [x] Follow-up for migrating the `release-template-operator-assets` pipeline created
  - this pipeline should be revisited as a whole - do we still need it? should the module template be there? 

### Attachments

Quickstart guide for Image Builder: https://github.com/kyma-project/test-infra/blob/main/cmd/image-builder/README.md#quickstart-guide

**Comment by amritanshusikdar** [comment]
The plan with this issue is to go phase-wise and not do everything at once. For the first phase, we have:

1. The GitHub Action workflow for image building has been added to [`template-operator`](https://github.com/kyma-project/template-operator/pull/243).
2. The respective [Prow Job](https://github.com/kyma-project/test-infra/pull/11766) in `test-infra` has been removed.
3. Assets uploading from GitHub Actions [here](https://github.com/kyma-project/template-operator/pull/245).

**Comment by amritanshusikdar** [comment]
As discussed in the daily, we renamed the newly introduced pipeline to `template-operator` from `pull-build-template-operator` to just `build-template-operator`, which can be seen [here](https://github.com/kyma-project/template-operator/pull/248).

Moreover, `main-template-op-build` and `release-template-operator-build` have also been removed. [See here](https://github.com/kyma-project/test-infra/pull/11768).
